### PR TITLE
chore(Composer): update to latest wpcs release

### LINKED ISSUE #37: Require latest version of WPCS always
Currently `composer.json` has a hard dependency on WPCS, and is using `0.10.0`. Because they're not at a 1.0 and the current rule is `^` this change will not be automatic for minor updates (only one a major lands will that be true). As a result, we need to update to the new standard manually.

### PR DESCRIPTION
**Pull request by ghost**
closes #37 

For those who want this now:

```json
{
  "repositories": [{
    "url": "https://github.com/jhabdas/wp-enforcer.git",
    "type": "git"
  }],
  "require-dev": {
    "stevegrunwell/wp-enforcer": "dev-master"
  }
}
```

**EDIT:** Probably don't use this until https://github.com/stevegrunwell/wp-enforcer/pull/38#issuecomment-302877536 is addressed.

### CODE REVIEW

**Comment by jhabdas** [comment]
Please merge directly to master. This project is not complex enough to warrant a dev branch IMO.

**Comment by ghost** [comment]
@stevegrunwell could you please switch the default branch back to `master` and update there. this current rig feels _sdrawkcab_?

**Comment by ghost** [comment]
Yep. I'm totally backed against the wrong branch in my fork due to the branch polarity reversal here. I'm going to push a hack up for now until I hear back before I rebase and update this pull to fix #39

File: bin/hooks/pre-commit [hunk 1]
```diff
@@ -14,9 +14,8 @@ fi
 # Run PHP_CodeSniffer
 echo "Running PHP_CodeSniffer..."
 
-git diff --name-only --cached --diff-filter=ACMRTUXB | xargs vendor/bin/phpcs $standard
-if [ $? != 0 ]
-then
+composer run lint:plugin
```
Comment (stevegrunwell) [inline bin/hooks/pre-commit:17]: Where is this command coming from? While it's handy to have a Composer script to run, I don't see a definition for this anywhere (nor is WP Enforcer only used for plugins, as this command seems to indicate).

**Comment by stevegrunwell** [comment]
> @stevegrunwell could you please switch the default branch back to master and update there. this current rig feels sdrawkcab?

The branching strategy isn't backwards, it may be just be a bit different than what you're used to (especially if you're typically using a "branch off of master, then merge into staging and/or master to deploy to staging or production, respectively" workflow).

[As outlined in the contributing doc](https://github.com/stevegrunwell/wp-enforcer/blob/develop/CONTRIBUTING.md#branching-strategy), new feature branches should come off of **develop** (which is the primary branch); the only time things are merged to master is when a new release is tagged, and that's by way of a `release/x.x.x` branch.

If you're not familiar with this pattern (commonly referred to as "Gitflow"), [I'd recommend checking out this documentation from Atlassian](https://www.atlassian.com/git/tutorials/comparing-workflows).

**Comment by ghost** [comment]
Closing this as the WP-CLI generates [`sample-plugin`](https://github.com/wp-cli/sample-plugin) containing a travis config which, for better or worse, does a `global` install of `wpcs` during CI, meaning it'll always get the latest build. As a result, it's probably better to set the version of `wpcs` to `*` to take upgrades as coding standards roll forward as opposed to seeing pulls like this which would simply cause noise in an already noisy world.

I'll open a bug for the related change, but I'm pulling back from using `wp-enforcer` myself as I find it confusing to directly depend on `wpcs` when it should really be a peer dependency.

**Pull request closed without merging**